### PR TITLE
walletdb: add sequence methods from bbolt to the main bucket interface

### PR DESCRIPTION
**Pull request by cfromknecht**
In this PR, we add the trio of sequence based methods that `bbolt` ships with to the main bucket interface. We do this in order to easily allow walletdb as is to be slotted into place where bbolt is currently used, without sacrificing any functionality.

Along the way we fix some issues with sub-modules that resulted in packges not being tested against the latest set of sub-modules. We also resolve an issue with `goclean.sh` that prevented any packages which were themselves sub-modules from being run.

### CODE REVIEW

File: walletdb/bdb/db.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (tx *transaction) ReadWriteBucket(key []byte) walletdb.ReadWriteBucket {
 }
 
 func (tx *transaction) CreateTopLevelBucket(key []byte) (walletdb.ReadWriteBucket, error) {
-	boltBucket, err := tx.boltTx.CreateBucket(key)
+	boltBucket, err := tx.boltTx.CreateBucketIfNotExists(key)
```
Comment (cfromknecht) [inline walletdb/bdb/db.go]: is this change relevant to the pr?

File: wtxmgr/tx_test.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func testDB() (walletdb.DB, func(), error) {
 	if err != nil {
 		return nil, func() {}, err
 	}
-	db, err := walletdb.Create("bdb", filepath.Join(tmpDir, "db"))
+	db, err := walletdb.Create("bdb", filepath.Join(tmpDir, "db"), true)
```
Comment (cfromknecht) [inline wtxmgr/tx_test.go]: should these be part of the prior commit?

File: walletdb/bdb/db.go (hunk 1 above)
Comment (Roasbeef) [reply walletdb/bdb/db.go]: Discussed offline that without this change, there isn't a way to conditionally obtain/create a top-level bucket.
